src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -257,9 +257,15 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
         /// Reusing <see cref="AssemblyMetadata"/> object allows for sharing data accross these references.
         /// </para>
         /// </remarks>
+        [Obsolete("The API is obsolete, use CreateFromFile(assembly.Location) instead", error: true)]
         public static MetadataReference CreateFromAssembly(Assembly assembly)
         {
-            return CreateFromAssembly(assembly, default(MetadataReferenceProperties));
+            throw new NotImplementedException();
+        }
```

Comment:
OK. Wonder if it's not better to break existing code now rather than when we move to CoreCLR though. I though the idea of this change is to make the transition of compilers to CoreCLR not breaking.